[std] Reword "cannot" in notes to not sound like negative permission (#6770)
cplusplus/draft: source/locales.tex

\begin{note}
Because facets are used by iostreams,
potentially while static constructors are running,

> it is not possible for their initialization to depend on programmed static initialization.

Reviewer: What is programmed static initialization?

Reviewer: I'll open an LWG issue for this as I have no idea what it's saying.

> their initialization would be unreliable if it depended on programmed static initialization.

Reviewer 2: ```suggestion
it is not possible for their initialization to depend on programmed static initialization.
```
